File: Examples/CSharp/Working-Shapes/RerouteConnectors.cs
using Aspose.Diagram;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Aspose.Diagram.Examples.CSharp.Working_Shapes
{
    public class RerouteConnectors
    {
        public static void Run()
        {
            // ExStart:RerouteConnectors
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Shapes();

            // Call a Diagram class constructor to load the VSDX diagram
            Diagram diagram = new Diagram(dataDir + "Drawing1.vsdx");
            // Get page by name
            Page page = diagram.Pages.GetPage("Page-3");

            // Get a particular connector shape
            Shape shape = page.Shapes.GetShape(18);
            // Set reroute option
            shape.Layout.ConFixedCode.Value = ConFixedCodeValue.NeverReroute;

            // Save Visio diagram
            diagram.Save(dataDir + "RerouteConnectors_Out.vsdx", SaveFileFormat.VSDX);
            // ExEnd:RerouteConnectors
        }
    }
}

File: Examples/VisualBasic/Working-Shapes/RerouteConnectors.vb

Imports Aspose.Diagram
Imports System

Public Class RerouteConnectors
    Public Shared Sub Run()
        ' ExStart:RerouteConnectors
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_Shapes()

        ' Call a Diagram class constructor to load the VSDX diagram
        Dim diagram As New Diagram(dataDir & Convert.ToString("Drawing1.vsdx"))
        ' Get page by name
        Dim page As Page = diagram.Pages.GetPage("Page-3")

        ' Get a particular connector shape
        Dim shape As Shape = page.Shapes.GetShape(18)
        ' Set reroute option
        shape.Layout.ConFixedCode.Value = ConFixedCodeValue.NeverReroute

        ' Save Visio diagram
        diagram.Save(dataDir & Convert.ToString("RerouteConnectors_Out.vsdx"), SaveFileFormat.VSDX)
        ' ExEnd:RerouteConnectors
    End Sub
End Class
